File: src/class/discord.js
const fetch = require('node-fetch')

class discord {
    constructor(key) {
        if (!key) throw new Error("No key provided")
        this.key = key
        this.baseUrl = "https://discord.com/api/v9/"
    }

    async sendMessage(channel, content) {
        let response = await fetch(`${this.baseUrl}channels/${channel}/messages`, {
            method: "POST",
            headers: {
                Authorization: 'Bot ' + this.key,
                "Content-Type": "application/json"
            },
            body: JSON.stringify({
                content,
                tts: false,
            })
        })

        response = await response.json()
        return response
    }

    async sendMultiPart(channel, form) {
        let response = await fetch(`${this.baseUrl}channels/${channel}/messages`, {
            method: 'POST',
            body: form,
            headers: { 
                Authorization: 'Bot ' + this.key
            }
        })

        response = await response.json()
        return response
    }

    async downloadAttachment(channel, attachment, filename) {
        const url = `https://cdn.discordapp.com/attachments/${channel}/${attachment}/${filename}`

        let response = await fetch(url, { method: 'GET' })
        if (!response.ok) {
            throw response.error()
        }
        response = await response.buffer()
        return response
    }

    async createChannel(channelName) {
        const response = await fetch(`${this.baseUrl}guilds/788450913136148600/channels`, {
            method: 'POST',
            headers: {
                Authorization: 'Bot ' + this.key,
                "Content-Type": "application/json"
            },
            body: JSON.stringify({
                name: channelName,
                type: 0,
                parent_id: '930204410850738176',
            })
        })

        const jsonRes = await response.json()
        return jsonRes
    }

    async deleteMessage(channel, message) {
        let response = await fetch(`${this.baseUrl}channels/${channel}/messages/${message}`, {
            method: 'DELETE',
            headers: {
                Authorization: 'Bot ' + this.key,
                "X-Audit-Log-Reason": 'Delete request by owner of the file'
            }
        })

        if (response.status != 204) {
            response = await response.json()
            return response
        } else {
            return 204
        }
    }

    async deleteChannel(channel) {
        let response = await fetch(`${this.baseUrl}channels/${channel}`, {
            method: 'DELETE',
            headers: {
                Authorization: 'Bot ' + this.key,
                "X-Audit-Log-Reason": 'Delete request by owner of the bucket'
            }
        })

        if (response.status != 204) {
            response = await response.json()
            return response
        } else {
            return 204
        }
    }
}

module.exports = discord

File: Discord/discord.go
package Discord

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"mime/multipart"
	"net/http"
	"net/textproto"
)

const channelID = "1037079784229978123"
const baseURL = "https://discord.com/api/v10"
const botToken = "Nzg4NDUyMDg4MTM2MjA0MzYx.GArHov.y0hV_prE0jSZeqzLDczf1Pbgd24c7CvAKoNKrs"

func SendMultipart(buffer []byte) (*Message, error) {
	var body bytes.Buffer

	writer := multipart.NewWriter(&body)
	writer.SetBoundary("boundary")

	header := textproto.MIMEHeader{}
	header.Add("Content-Disposition", "form-data; name=\"payload_json\"")
	header.Set("Content-Type", "application/json")
	jsonWriter, err := writer.CreatePart(header)
	if err != nil {
		return nil, err
	}

	payload := map[string]interface{}{
		"attachments": []interface{}{map[string]interface{}{
			"id":          0,
			"description": "Part of a file",
			"filename":    "part.fbin",
		}},
	}
	payloadBytes, err := json.Marshal(payload)
	if err != nil {
		return nil, err
	}

	jsonWriter.Write(payloadBytes)

	bufferWriter, err := writer.CreateFormFile("files[0]", "part.fbin")
	if err != nil {
		return nil, err
	}

	bufferWriter.Write(buffer)

	err = writer.Close()
	if err != nil {
		return nil, err
	}

	url := fmt.Sprintf("%s/channels/%s/messages", baseURL, channelID)
	request, err := http.NewRequest("POST", url, &body)
	//fmt.Println("Multipart created!", request)
	if err != nil {
		return nil, err
	}

	request.Header.Set("Content-Type", writer.FormDataContentType())
	request.Header.Set("authorization", fmt.Sprintf("Bot %s", botToken))

	//fmt.Println(request.URL)

	client := http.Client{}
	response, err := client.Do(request)
	if err != nil {
		return nil, err
	}

	if response.StatusCode == http.StatusOK {
		//fmt.Println("Status OK")
		var result Message
		err := json.NewDecoder(response.Body).Decode(&result)
		if err != nil {
			return nil, err
		}

		return &result, nil
	} else {
		body, _ := io.ReadAll(response.Body)
		return nil, fmt.Errorf("error uploading file. status: %s\nresponse Body: %s", response.Status, body)
	}
}
